src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp

```diff
@@ -610,6 +605,82 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
         }
     }
 
+    /**
+     * @brief Checks and translates package information.
+     *
+     * This function searches for translation data in the Level 1 and Level 2 caches for a given package.
+     * If translation data is found, it populates the translations and returns them. If no translation data
+     * is found, it logs a debug message and returns an empty vector.
+     *
+     * @param package The package data to be checked and translated.
+     * @param osPlatform The operating system platform for which the package translation is required.
+     * @return A vector containing the translated package data.
+     */
+    std::vector<PackageData> checkAndTranslatePackage(const PackageData& package, const std::string& osPlatform)
+    {
+        std::vector<PackageData> vulnerabilityTranslations;
+        const auto cacheKey = osPlatform + "_" + package.vendor + "_" + package.name;
+
+        auto translatePackage = [&](const auto& translations)
+        {
+            for (const auto& translation : translations)
+            {
+                PackageData translatedPackage = package;
+                if (!translation.name.empty())
+                {
+                    translatedPackage.name = translation.name;
+                }
+                if (!translation.vendor.empty())
+                {
+                    translatedPackage.vendor = translation.vendor;
+                }
+                if (!translation.version.empty())
+                {
+                    translatedPackage.version = translation.version;
+                }
+                vulnerabilityTranslations.push_back(translatedPackage);
+            }
+        };
+
```

Comment:
Improve (readability and design):
1. Check if the translation is in the filter.
    - If it's in the filter, log this information and return the empty translation
2. Check L1 (if not in the filter)
3. Check L2 (if not in L1)
4. Add to the filter (if not in L2)